yt/frontends/ramses/io.py

```diff
@@ -278,6 +278,10 @@ def _read_fluid_file_descriptor(fname):
         ('pressure', 'Pressure'),
         ('metallicity', 'Metallicity'),
     ]
+
+    #Magnetic field file descriptors
+    magnetic=np.array([['B_{0}_{1}'.format(dim,side) for side in ['left','right']] for dim in ['x','y','z']]).ravel()
```

Comment:
Since you are only using `magnetic` below to find varnames, it may be better to use a simple python list instead. You can build it using
```suggestion
    magnetic = ['B_{0}_{1}'.format(dim,side) for side in ['left','right'] for dim in ['x','y','z']]
```

Reply:
Also if we want to write a code that would be consistent with the `mapping` above, you could rewrite it as follows:
```python
magnetic_mapping = [(key, key) for key in ('B_{0}_{1}'.format(dim,side) for side in ['left','right'] for dim in ['x','y','z'])]
mapping += magnetic_mapping
```
and remove bellow the lines reading
```python
elif varname in magnetic:
    varname = varname
```